crates/aiken-lang/src/gen_uplc/air.rs

```diff
@@ -50,6 +50,13 @@ pub enum Air {
         recursive_nonstatic_params: Vec<String>,
         variant_name: String,
     },
+    DefineCyclicFuncs {
+        func_name: String,
+        module_name: String,
+        variant_name: String,
+        // just the params
+        contained_functions: Vec<Vec<String>>,
+    },
```

Comment:
I am curious if there's perhaps a way to unify `DefineFunc` and `DefineCyclicFuncs` so that both can be handled in one holistic way.

Reply:
As long as it's not more complicated! Sometimes trying to use one thing for two things makes things a pain.

Reply:
Yeah unfortunately it's more than just a toggle. Each mutually recursive function needs to keep track of its own parameters. And at the top level we have a function with 2 params (self function param and index param)